source/common/network/io_socket_handle_impl.h

```diff
@@ -96,7 +96,7 @@ class IoSocketHandleImpl : public IoHandle, protected Logger::Loggable<Logger::I
       return Api::IoCallUint64Result(result.return_value_,
                                      Api::IoErrorPtr(nullptr, IoSocketError::deleteIoError));
     }
-    RELEASE_ASSERT(result.errno_ != SOCKET_ERROR_INVAL, "Invalid argument passed in.");
+    ENVOY_BUG(result.errno_ != SOCKET_ERROR_INVAL, "Invalid argument passed in.");
```

Comment:
As discussed offline, let's make this an ENVOY_LOG error?

Reply:
done